grc/business_logic/data_structures/application_data.py

```diff
@@ -154,3 +167,10 @@ def section_status_submit_and_pay_data(self) -> ListStatus:
                     return ListStatus.IN_PROGRESS
 
         return ListStatus.IN_REVIEW
+
+
+def any_duplicate_aws_file_names(uploads_files: List[EvidenceFile]):

```

Comment:
`def any_duplicate_aws_file_names(uploads_files: List[EvidenceFile]):
    files = [file.aws_file_name for file in uploads_files]
    return files == [file for file in files if files.count(file) > 1]`

Is much more concise

Reply:
I think...
```python
files == [file for file in files if files.count(file) > 1]
```
...would only return True if **all** the files were duplicates, since the `==` checks that the arrays contain exactly matching elements.

I've gone with...
```python
def any_duplicate_aws_file_names(uploads_files: List[EvidenceFile]):
    aws_file_names = [file.aws_file_name for file in uploads_files]
    duplicate_aws_file_names = [file_name for file_name in aws_file_names if aws_file_names.count(file_name) > 1]
    any_duplicates = len(duplicate_aws_file_names) > 0
    return any_duplicates
```
...which I'm hoping is very clear (and debuggable), even if it's a line or 2 longer than it strictly could be